tsdb/engine/tsm1/compact_test.go

```diff
@@ -2482,6 +2485,250 @@ func TestDefaultPlanner_FullyCompacted_SmallSingleGeneration_Halt(t *testing.T)
+
+	cgroup, cgLen, genLen := cp.PlanOptimize()
+	require.Equal(t, []tsm1.CompactionGroup(nil), cgroup)
+	require.Equal(t, int64(0), cgLen)
+	require.Equal(t, int64(0), genLen)
+}
+
+// This test is added to account for a single generation that has a group size
+// over 2 GB at max points per block with aggressive compaction, and, 1 file
+// under 2 GB at default max points per block.
+// It should not compact any further.
+func TestDefaultPlanner_FullyCompacted_LargeSingleGenerationNoMaxAggrBlocks(t *testing.T) {
+	// > 2 GB total group size
+	// 100% of files are at aggressive max block size
+	data := []tsm1.FileStat{
+		{
+			Path: "01-13.tsm1",
+			Size: 2048 * 1024 * 1024,
+		},
+		{
+			Path: "01-14.tsm1",
+			Size: 691 * 1024 * 1024,
+		},
+	}
+
+	fs := &fakeFileStore{
+		PathsFn: func() []tsm1.FileStat {
+			return data
+		},
+	}
+	blocks := []int{
+		tsdb.AggressiveMaxPointsPerBlock,
+		tsdb.DefaultMaxPointsPerBlock,
+	}
+	err := fs.SetBlockCounts(blocks)
+	require.NoError(t, err)
+
+	cp := tsm1.NewDefaultPlanner(fs, tsdb.DefaultCompactFullWriteColdDuration)
+
+	compacted, reason := cp.FullyCompacted()
+	reasonExp := ""
+	require.Equal(t, reason, reasonExp)
+	require.Equal(t, true, compacted)
+
+	cgroup, cgLen, genLen := cp.PlanOptimize()
+	require.Equal(t, []tsm1.CompactionGroup(nil), cgroup)
+	require.Equal(t, int64(0), cgLen)
+	require.Equal(t, int64(0), genLen)
+}
+
+// This test is added to account for a single generation that has a group size
+// over 2 GB and multiple files under 2 GB all at max points per block for aggressive compaction.
+func TestDefaultPlanner_FullyCompacted_ManySingleGenLessThen2GBMaxAggrBlocks(t *testing.T) {
+	// > 2 GB total group size
+	// 100% of files are at aggressive max block size
+	data := []tsm1.FileStat{
+		{
+			Path: "01-13.tsm1",
+			Size: 2048 * 1024 * 1024,
+		},
+		{
+			Path: "01-14.tsm1",
+			Size: 650 * 1024 * 1024,
+		},
+		{
+			Path: "01-15.tsm1",
+			Size: 450 * 1024 * 1024,
+		},
+	}
+
+	fs := &fakeFileStore{
+		PathsFn: func() []tsm1.FileStat {
+			return data
+		},
+	}
+	blocks := []int{
+		tsdb.AggressiveMaxPointsPerBlock,
+		tsdb.AggressiveMaxPointsPerBlock,
+		tsdb.AggressiveMaxPointsPerBlock,
+	}
+	err := fs.SetBlockCounts(blocks)
+	require.NoError(t, err)
+
+	cp := tsm1.NewDefaultPlanner(fs, tsdb.DefaultCompactFullWriteColdDuration)
+
+	compacted, reason := cp.FullyCompacted()
+	reasonExp := ""
+	require.Equal(t, reason, reasonExp)
+	require.Equal(t, true, compacted)
+
+	cgroup, cgLen, genLen := cp.PlanOptimize()
+	require.Equal(t, []tsm1.CompactionGroup(nil), cgroup)
+	require.Equal(t, int64(0), cgLen)
+	require.Equal(t, int64(0), genLen)
+}
+
+// This test is added to account for a single generation that has a group size
+// over 2 GB and multiple files under 2 GB with multiple files under aggressive
+// max points per block. This should further compact.
+func TestDefaultPlanner_FullyCompacted_ManySingleGenLessThen2GBNotMaxAggrBlocks(t *testing.T) {
```

Comment:
Do you want to call `PlanOptimize` in any of there tests of `FullyCompacted` to verify it is producing the correct plan?

Reply:
There is a test where I do call PlanOptimize() but I can call it in more of the tests for FullyCompacted 👍

Reply:
Basically, run the code [here](https://github.com/influxdata/influxdb/blob/db523227a2c8d8a05e317dbe0f1b9630b05838c6/tsdb/engine/tsm1/engine.go#L2126C4-L2137C1) in every test and ensure the outputs are as expected.